python/cuspatial/cuspatial/core/geoseries.py

```diff
@@ -517,3 +509,102 @@ def to_arrow(self):
                 arrow_polygons,
             ],
         )
+
+    def _align_to_index(
+        self: T,
+        index: ColumnLike,
+        how: str = "outer",
+        sort: bool = True,
+        allow_non_unique: bool = False,
+    ) -> T:
+        """
+        The values in the newly aligned columns will not change,
+        only positions in the union offsets and type codes.
+        """
+        aligned_union_offsets = (
+            self._column._meta.union_offsets._align_to_index(
+                index, how, sort, allow_non_unique
+            )
+        ).astype("int32")
+        aligned_union_offsets[
+            aligned_union_offsets.isna()
+        ] = Feature_Enum.NONE.value
+        aligned_input_types = self._column._meta.input_types._align_to_index(
+            index, how, sort, allow_non_unique
+        ).astype("int8")
+        aligned_input_types[
+            aligned_input_types.isna()
+        ] = Feature_Enum.NONE.value
+        column = GeoColumn(
+            (
+                self._column.points,
+                self._column.mpoints,
+                self._column.lines,
+                self._column.polygons,
+            ),
+            {
+                "input_types": aligned_input_types,
+                "union_offsets": aligned_union_offsets,
+            },
+        )
+        return GeoSeries(column)
+
+    def align(self, other):
+        """
+        Align the rows of two GeoSeries using outer join.
```

Comment:
What does it mean to align the rows of geoseries? I think a more detailed explanation would help.

Reply:
I will write one that covers this implementation.

Reply:
I increased the docs, explaining three use cases, and provided examples for them. Please approve and merge @harrism .